src/core/solvers/box-qp.cpp

```diff
@@ -0,0 +1,258 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2019-2020, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#include <iostream>
+#include "crocoddyl/core/solvers/box-qp.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+
+BoxQP::BoxQP(const std::size_t nx, std::size_t maxiter, const double th_acceptstep, const double th_grad,
+             const double reg)
+    : nx_(nx),
+      maxiter_(maxiter),
+      th_acceptstep_(th_acceptstep),
+      th_grad_(th_grad),
+      reg_(reg),
+      x_(nx),
+      xnew_(nx),
+      g_(nx),
+      dx_(nx) {
+  // Check if values have a proper range
+  if (0. >= th_acceptstep && th_acceptstep >= 0.5) {
+    std::cerr << "Warning: th_acceptstep value should between 0 and 0.5" << std::endl;
+  }
+  if (0. > th_grad) {
+    std::cerr << "Warning: th_grad value has to be positive." << std::endl;
+  }
+  if (0. > reg) {
+    std::cerr << "Warning: reg value has to be positive." << std::endl;
+  }
+
+  // Reserve the space and compute alphas
+  clamped_idx_.reserve(nx_);
+  free_idx_.reserve(nx_);
+  const std::size_t& n_alphas_ = 10;
+  alphas_.resize(n_alphas_);
+  for (std::size_t n = 0; n < n_alphas_; ++n) {
+    alphas_[n] = 1. / pow(2., static_cast<double>(n));
```

Comment:
This version of pow is inefficient, but it might not be a serious problem as it is called once.

Reply:
Yes, I preferred to use it (for code readability) since it's needed only at config-time.